### PR TITLE
add bathroom scale

### PR DESCRIPTION
**Pull request by KorGgenT**
#### Summary

```SUMMARY: Features "Add bathroom scale iuse"```
#### Purpose of change

I wanted a more accessible way for the player to see their weight.
#### Describe the solution

Created a new iuse_actor called weigh_self_actor
created a basic bathroom scale item.
max_weight is in grams. shows "ERROR" if you are too heavy.
Note: I did not add this to itemgroups, if someone else would be so kind to do that it would be much appreciated.

image

Shows kgs or lbs depending on what setting you have.

### CODE REVIEW

**Comment by NickGrease** [comment]
With analog scales you'd know you were at least xxx kg/lb's (the dial would stop at what ever the max weight it could handle). Never used a digital one - but I think it would be fair to include the max weight in the error message, "ERROR Max Weight (xxx) Exceeded" or some such. Just so the person has at least some idea of their weight.

File: src/iuse_actor.cpp [hunk 1]
```diff
@@ -4243,3 +4243,32 @@ bool deploy_tent_actor::check_intact( const tripoint &center ) const
...
+    dump.emplace_back( "DESCRIPTION",
+                       _( "Use this item to weigh yourself.  Includes everything you are wearing." ) );
+}
+
+int weigh_self_actor::use( player &p, item &, bool, const tripoint & ) const
+{
+    // this is a weight, either in kgs or in lbs.
+    double weight = convert_weight( p.get_weight() );
+    if( weight > convert_weight( max_weight ) ) {
+        popup( "ERROR TOO HEAVY" );
```
Comment (Night-Pryanik) [inline src/iuse_actor.cpp]:
```suggestion
        popup( _( "ERROR TOO HEAVY" ) );
```

File: src/iuse_actor.cpp [hunk 2]
```diff
@@ -4243,3 +4243,32 @@ bool deploy_tent_actor::check_intact( const tripoint &center ) const
...
+    } else {
+        popup( string_format( "%.0f %s", weight,
+                              get_option<std::string>( "USE_METRIC_WEIGHTS" ) == "lbs" ? _( "lbs" ) : _( "kgs" ) ) );
```
Comment (Night-Pryanik) [inline src/iuse_actor.cpp]: We have a `weight_units()` function for this.

File: src/iuse_actor.cpp (hunk 1 above)
Comment (Mecares) [reply src/iuse_actor.cpp]: Wouldn't be "ERROR: TOO HEAVY" be better than "ERROR TOO HEAVY"?

File: src/iuse_actor.cpp (hunk 2 above)
Comment (KorGgenT) [reply src/iuse_actor.cpp]: thanks!

File: src/iuse_actor.cpp (hunk 1 above)
Comment (KorGgenT) [reply src/iuse_actor.cpp]: i added the translate string and modified the message to include a colon and the max weight.

File: src/iuse_actor.cpp [hunk 3]
```diff
@@ -4255,10 +4255,9 @@ int weigh_self_actor::use( player &p, item &, bool, const tripoint & ) const
     // this is a weight, either in kgs or in lbs.
     double weight = convert_weight( p.get_weight() );
     if( weight > convert_weight( max_weight ) ) {
-        popup( "ERROR TOO HEAVY" );
+        popup( _( "ERROR: Max weight of %.0f %s exceeded" ), weight, weight_units() );
```
Comment (NickGrease) [inline src/iuse_actor.cpp]: Shouldn't this be **max_weight**? - the idea being that they wouldn't know their actual weight, but they would know its over what ever the maximum the scale can handle.

Comment (NickGrease) [inline src/iuse_actor.cpp]: Shouldn't this be max_weight? - the idea being that they wouldn't know their actual weight, but they would know its over what ever the maximum the scale can handle.

Comment (KorGgenT) [reply src/iuse_actor.cpp]: ahah, you're right

**Comment by int-ua** [comment]
FYI: ASCII fallback for ■ symbol doesn't work with current tilesets, it's becoming unknown tile.

**Pull request merged**